logprep/connector/confluent_kafka/input.py

```diff
@@ -504,11 +504,20 @@ def _assign_callback(self, consumer, topic_partitions):
             self.metrics.current_offsets.add_with_labels(offset, labels)
 
     def _revoke_callback(self, consumer, topic_partitions):
+
+        if getattr(consumer, "closed", False):
+            raise RuntimeError("Cannot revoke: consumer is already closed")
+
         for topic_partition in topic_partitions:
             self.metrics.number_of_warnings += 1
+            try:
+                member_id = consumer.memberid()
+            except Exception as error:
+                raise RuntimeError(f"Failed to retrieve member ID: {error}\n") from error
+
             logger.warning(
                 "%s to be revoked from topic: %s | partition %s",
-                consumer.memberid(),
+                member_id,
                 topic_partition.topic,
                 topic_partition.partition,
             )
```

Comment:
```suggestion
        for topic_partition in topic_partitions:
            self.metrics.number_of_warnings += 1
            member_id = self._get_memberid()

            logger.warning(
                "%s to be revoked from topic: %s | partition %s",
                member_id,
                topic_partition.topic,
                topic_partition.partition,
            )
```

Consider to except only, cause asking for permission ("is consumer closed?") and then also asking for forgiveness ("give me the memeberid ... ups... consumer is already closed") seems a bit redundant. 

Additionally consider to refactor the forgiveness part to this method:

```python
    def _get_memberid(self):
        try:
            member_id = self.consumer.memberid()
        except RuntimeError as error:
            logger.error("Failed to retrieve member ID: %s", error)
        return member_id
```

and use it in the `_assign_callback` and in the `_lost_callback` also, so that you don't repeat code.

Reply:
Added to the code